### PR TITLE
DO NOT MERGE [DEVELOPER-5432] Assembly Code Snippets

### PR DESCRIPTION
**Pull request by jordanpagewhite**
This implements a new code snippet assembly type and uses Highlight.js
for code syntax highlighting. Some remaining tasks include determining a
syntax highlighting theme that is on-brand and attractive.

### JIRA Issue Link
* https://issues.jboss.org/browse/DEVELOPER-5432

### Verification Process

Go here:

Relative URL: `/node/add/article`

Do this:

* Enter any content for required fields
* Once you reach the Content field, you should notice that you now have an option of using a Code Snippet assembly type.
* Add a new code snippet assembly type
* Enter some name for the code snippet that makes sense to you
* In the Code, long text unformatted field, enter some code like:

```
var msg = 'hello world';
console.log(msg);
```

* Finish filling out any required fields and save the edit form to create the Article

You should see this:

* You should be redirected to your newly created Article containing a code snippet assembly

githubhighlightjstheme

### CODE REVIEW

**Comment by jordanpagewhite** [comment]
Gina requested we use the Github HighlightJS theme, and I made those changes in my last commit: https://issues.jboss.org/browse/DEVELOPER-5432

**Comment by LightGuard** [comment]
@jordanpagewhite is this still a work in progress?

**Comment by jordanpagewhite** [comment]
@LightGuard Nope, this is ready to go. I updated the title. Thank you for asking.

**Comment by LightGuard** [comment]
Would like to get Alloy's take on this, but it looks good to me.

**Comment by rogeralloymagnetic** [comment]
> Would like to get Alloy's take on this, but it looks good to me.

Code looks good to me. @fearlsgroove any opinions?

File: _docker/drupal/drupal-filesystem/web/themes/custom/rhdp/rhdp.theme [hunk 1]
```diff
@@ -512,4 +512,14 @@ function rhdp_preprocess_assembly(&$variables) {
       $eloqua_assembly_id = $assembly->id();
       $variables['eloqua_assembly_id'] = $eloqua_assembly_id;
     }
+
+    if ($assembly->hasField('field_code')) {
+      $field_code = $assembly->get('field_code')->getValue()[0]['value'];
+	  $variables['code'] = preg_replace(
+		'/(<pre><code>(.)+<\/code><\/pre>)/',
+		'',
```
Comment (fearlsgroove) [inline _docker/drupal/drupal-filesystem/web/themes/custom/rhdp/rhdp.theme]: I feel like regexp replacements is always a sure sign of trouble. Why is this bit necessary? Can we accomplish whatever this is after in a different way?

**Comment by rogeralloymagnetic** [comment]
@jordanpagewhite Also, all assemblies need the existing `field_audience_selection` field added.

**Comment by jordanpagewhite** [comment]
Thank you @rogeralloymagnetic and @fearlsgroove . I will take care of these changes today.

**Pull request closed without merging**